### PR TITLE
Various warning fixes for review...

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: libsrc2/hyper.c [hunk 1]
```diff
@@ -78,10 +78,10 @@ offset_to_index(int ndims,
 
 /** The main restructuring code.
  */
-void restructure_array(int ndims,    /* Dimension count */
+static void restructure_array(hsize_t ndims,    /* Dimension count */
```
Comment (rdvincent) [inline libsrc2/hyper.c]: Hi - unfortunately restructure_array() is used in one of the MINC tools, so it must remain public. If there is a warning you're seeing that this fixes, we need to find a better solution.

**Comment by seanm** [comment]
It's the same 'missing prototype' warning... should restructure_array be declared in a .h then?

**Comment by rdvincent** [comment]
Of course. I'll do that today.

**Comment by vfonov** [comment]
Linking C executable copy_xfm
../libminc2.so.2.3.00: undefined reference to `restructure_array'

**Comment by rdvincent** [comment]
Yes, I know. I will check in a fix in a few minutes.

**Comment by vfonov** [comment]
By the way, restructure_array is used in https://github.com/BIC-MNI/libminc/blob/master/libsrc/minc_simple.c#L407 

which is inside MINC1, so if one decides to build libminc without MINC2 support this will probably fail.

**Comment by rdvincent** [comment]
I'm working on exactly that.

**Pull request merged**